src/VolumeCache.ts

```diff
@@ -0,0 +1,306 @@
+type XYZ<T> = { x: T; y: T; z: T };
+export type VolumeScaleDims = XYZ<number>;
+type CacheEntryExtent = XYZ<[number, number]>;
+
+// The following two very similar types are kept separate because we may later want to allow more
+// complex queries with respect to scale, e.g. "get the largest available scale within this range"
+export type DataArrayExtent = CacheEntryExtent & {
+  scale: number;
+  time: number;
+  channel: number;
+};
+
+export type QueryExtent = CacheEntryExtent & {
+  scale: number;
+  time: number;
+};
+
+type CacheEntry = {
+  /** The data contained in this entry */
+  // TODO allow more types of `TypedArray` to be stored together in the cache?
+  data: Uint8Array;
+  /** The subset of the volume covered by this entry */
+  size: CacheEntryExtent;
+  /** The previous entry in the LRU list */
+  prev: MaybeCacheEntry;
+  /** The next entry in the LRU list */
+  next: MaybeCacheEntry;
+  /** The array which contains this entry */
+  parentArr: CacheEntry[];
+};
+type MaybeCacheEntry = CacheEntry | null;
```

Comment:
suggest moving this line up to just after where CacheEntry was defined

Reply:
like immediately before?
```javascript
type MaybeCacheEntry = CacheEntry | null;
type CacheEntry = {
// ...
```

Reply:
omg, I didn't even realize I was reading the definition of CacheEntry.  It's the recursion that got me.  nevermind, this is a nonissue. sorry!

Reply:
the thing is, I kind of like it going there instead... puts both names right next to each other so a reader can more easily pick up on their relationship